src/enclave/enclave_timer.c

```diff
@@ -28,7 +28,11 @@ _Atomic(uint64_t) internal_counter = 0;
  */
 uint64_t enclave_nanos()
 {
-    uint64_t e = sgxlkl_enclave_state.shared_memory.timer_dev_mem->nanos;
+    struct timer_dev* t = sgxlkl_enclave_state.shared_memory.timer_dev_mem;
+
+    sgxlkl_ensure_outside(t, sizeof(struct timer_dev));
```

Comment:
Should do the memory check when intialize sgxlkl_enclave_state.shared_memory, instead of here, to avoid unncessary performace waste.

Reply:
Yes.